.buildkite/pipeline.fleet-server-tests.yaml

```diff
@@ -0,0 +1,25 @@
+# This pipeline serves as the entry point for your service's quality gates definitions. When
+# properly configured, it will be invoked automatically as part of the automated
+# promotion process once a new version was rolled out in one of the various cloud stages.
+#
+# The updated environment is provided via ENVIRONMENT variable. The seedling
+# step will branch and execute pipeline snippets at the following location:
+# .buildkite/pipeline.tests-qa.yaml
+# .buildkite/pipeline.tests-staging.yaml
+# .buildkite/pipeline.tests-production.yaml
+#
+# Docs: https://docs.elastic.dev/serverless/qualitygates
+
+env:
+  ENVIRONMENT: ${ENVIRONMENT}
```

Comment:
```suggestion
  ENVIRONMENT: ${ENVIRONMENT?}
```

Reply:
This was causing [issues](https://github.com/elastic/fleet-server/pull/2851) before on main when triggered without ENVIRONMENT. How can we prevent this pipeline from being run without ENVIRONMENT?

Reply:
This pipeline has to have a Target Environment for its tests, 
otherwise it has to fail. 
And the earlier the better. 
It is the task of any invoker to guarantee that the ENVIRONMENT is set.
When you run manaul builds, it should be handed over via run settings.

But I am not sure If I am following. 
This should NOT run on PRs or main merges or any github actions at all. I think we need to add [after this](https://github.com/elastic/fleet-server/blob/22832f2b17cafd1891952d3c19ef80432b1f24c9/catalog-info.yaml#L134).
```
      provider_settings:
        trigger_mode: none
```